src/qtfirebaseadmob.cpp

```diff
@@ -995,7 +995,7 @@ firebase::FutureBase QtFirebaseAdMobInterstitial::initInternal()
     return firebase::FutureBase(); // invalid future because we already handled it
 }
 
-void QtFirebaseAdMobInterstitial::onPresentationStateChanged(int state)
+void QtFirebaseAdMobInterstitial::onPresentationStateChanged(PresentationState state)
```

Comment:
AFAIK - you can't pass custom types to `QML` - that's why it's passed as `int` - Does this work from QML?

Reply:
Q_ENUMS(enumname) makes the enum available in Qml.

Reply:
Yeah but what about the `PresentationState state` argument - I don't think / didn't know you can pass custom types as arguments?

That's why we use `int` [here](https://github.com/Larpon/QtFirebase/blob/14a6a920/src/qtfirebaseadmob.h#L240) - I might be wrong - as it's some time since I've had my nose in it - but I'm pretty sure that's why we use int.

[In this QML](https://github.com/Larpon/QtFirebaseExample/blob/79474c9/App/qml/pages/AdMob.qml#L163) from the example app the code was previously `undefined` because we passed it as the ENUM type `Error`.

You can see the commit [here](https://github.com/Larpon/QtFirebase/commit/757343623644b23b12dd2d3ae502d90e07ebe668#diff-789c2f7c561436c12379919964d08e46L273)

Reply:
For me this works. I am using it all the time in our projects.

Reply:
Cool - I'll have a go at it when get time

Reply:
Okay - I gave it a spin - right now I can't do this in QML:
```
AdMobInterstitial {
        ...

        onPresentationStateChanged: { // <- This gives an error: "Cannot assign to non-existent property "onPresentationStateChanged""
            
        }
    }
```

Which is very logical as we can only connect to `signals` - not `slots` ... I'm not pulling this in, sorry

Reply:
```
AdMobInterstitial {
        ...

        onPresentationStateChanged: {
            // This gives an error: Cannot assign to non-existent property "onPresentationStateChanged"
        }
    }
```

Reply:
You are right - I oversaw this. I will upload a fix this afternoon.